app/components/Settings/Preview/TableTree.react.js

```diff
@@ -78,45 +167,18 @@ class TableTree extends Component {
     render() {
         const schemaRequest = this.props.schemaRequest || {};
         const status = schemaRequest.status;
-
-        if (typeof status === 'undefined' || status === 'loading') {
-            return (<div className="loading">{'Loading'}</div>);
-        }
-
-        if (status !== 200) {
-            return (
-                <div style={{padding: '5px', fontSize: '12px'}}>
-                    {`ERROR ${JSON.stringify(schemaRequest)}`}
-                </div>
-            );
-        }
-
         const preview = this.props.preview || {};
         const treeSchema = preview.treeSchema;
 
-        if (!treeSchema) {
-            return (<div className="loading">{'Updating'}</div>);
-        }
+        this.isConnecting(status, schemaRequest, treeSchema);
```

Comment:
This line breaks `<TableTree>`. Please don't move this code into `isConnecting`:
- the logic to check whether Falcon is still connecting is just one line `(typeof status === 'undefined' || status === 'loading')`
- the logic to check whether there was an error is also one line `(status !== 200)`
- functions named `isXXX` should return a boolean.